src/SSVOpenHexagon/Components/CWall.cpp

```diff
@@ -55,23 +55,34 @@ void CWall::update(HexagonGame& mHexagonGame, ssvu::FT mFT)
 void CWall::moveTowardsCenter(
     HexagonGame& mHexagonGame, const sf::Vector2f& mCenterPos, ssvu::FT mFT)
 {
+    const float wallSpawnDist{mHexagonGame.getLevelStatus().wallSpawnDistance};
+    const float wallAccSpawnDist{mHexagonGame.getLevelStatus().wallAccSpawnDistance};
+
     const float radius{mHexagonGame.getRadius() * 0.65f};
+    const float outerBounds =
+        (wallSpawnDist == wallAccSpawnDist) ? wallSpawnDist * 0.5f : 800.f;
 
     int pointsOnCenter{0};
+    int pointsOutOfBounds{0};
     for(sf::Vector2f& vp : vertexPositions)
     {
         if(std::abs(vp.x - mCenterPos.x) < radius &&
             std::abs(vp.y - mCenterPos.y) < radius)
         {
             ++pointsOnCenter;
-        }
+        } 
         else
         {
+			if(std::abs(vp.x - mCenterPos.x) > outerBounds &&
+                std::abs(vp.y - mCenterPos.y) > outerBounds)
+			{
+				++pointsOutOfBounds;
+			}
             ssvs::moveTowards(vp, mCenterPos, speed.speed * 5.f * mFT);
         }
     }
 
-    if(pointsOnCenter > 3)
+    if((pointsOnCenter > 3 || pointsOutOfBounds > 3) || (pointsOnCenter == 2 && pointsOutOfBounds == 2))
```

Comment:
This needs a comment in the code, not sure why it is the way it is

Reply:
The first half of the condition will destroy the wall if it is in the center or out of bounds. The second half addresses an edge case where a wall is so thick it can reach the center and out of bounds.